feat: onchainkit initial commit

rin-st opened the pull request:
Adds support of [OnchainKIt](https://onchainkit.xyz/)
Adds page with OnchainKit components

Fixes https://github.com/scaffold-eth/create-eth-extensions/issues/8

Notes:
See Readme for networks support
Our react-query, viem and wagmi versions should be very close or even the same as in OnchainKit, or multiple error occurs. Not sure how to make it up to date all the time
[Tailwind integrations](https://onchainkit.xyz/guides/tailwind) requires to define theme variables, and some of them overwrite ours (works for other projects with tailwind)

To test
```
yarn cli -e scaffold-eth/create-eth-extensions:onchainkit-init
```

TODO:
- [ ] Update `create-eth` first: https://github.com/scaffold-eth/create-eth/pull/74

technophile-04 commented:
Ohh while testing https://github.com/scaffold-eth/create-eth/pull/74

I was getting when I navigate to /onchainkit-examples : 
![image](https://github.com/scaffold-eth/create-eth-extensions/assets/80153681/1496d981-b3df-4c01-816c-fa95128f4a26)

And also theme behaves in different way : 

![Screenshot 2024-07-09 at 3 58 38 PM](https://github.com/scaffold-eth/create-eth-extensions/assets/80153681/e97868bf-8228-4b61-be2d-42606d108043)

rin-st commented:
> I was getting when I navigate to /onchainkit-examples :

Updated version, now it should work. It's because of 
> Our react-query, viem and wagmi versions should be very close or even the same as in OnchainKit, or multiple error occurs.

I didn't know when that when merging `package.json`s , `merge-packages` doesn't just take lib from second package's when versions are different but tries to find intersection first. I'll add pr to mention it later

----
> And also theme behaves in different way :

It's because of 
> [Tailwind integrations](https://onchainkit.xyz/guides/tailwind) requires to define theme variables, and some of them overwrite ours (works for other projects with tailwind)

Removed `secondary` related vars and extends, so fills works fine. But not sure onchainkit will have necessary colors after it 🤷‍♂️ .

---

when testing don't forget to get onchainkit API key (see readme)

Thanks!

technophile-04 commented:
Thanks Rinat! Also noticed that when you do `yarn next:check-types` there are some error related to address not being assignable to string maybe due to `moduleResolution: "bundler"`?

rin-st commented:
Updated https://github.com/scaffold-eth/create-eth/pull/74 and this pr. Everything seems work and no errors during `yarn next:check-types`

technophile-04 commented on extension/packages/nextjs/components/OnchainKitScaffoldProvider.tsx:
```diff
@@ -0,0 +1,15 @@
+import { useAccount } from "wagmi";
+
+export const OnchainKitScaffoldProvider = ({ children }: PropsWithChildren) => {
+  const { chain } = useAccount();
+
+  return chain ? (
```
Maybe instead of returning empty children, what if we just default the chain to base? 

For example : 

```

  return (
    <OnchainKitProvider chain={chain ? chain : base} apiKey={process.env.NEXT_PUBLIC_ONCHAINKIT_API_KEY}>
      {children}
    </OnchainKitProvider>
  );
};
```

rin-st replied:
Thanks! Updated [fd82574](https://github.com/scaffold-eth/create-eth-extensions/pull/11/commits/fd82574b82887ee757129487f71b1418e5dc03ca)

technophile-04 commented:
Thanks @rin-st!! Seems to work great! Will try to test / review tomorrow, Also I am not sure whats the best way to handle this but at some places our SE-2 colors are changed for example : 

![Screenshot 2024-07-22 at 11 30 57 PM](https://github.com/user-attachments/assets/d7185fc0-4489-468e-b18a-6283935163a3)

Notice it has changed it color to violet when you hover over the Switch Icon

rin-st commented:
Updated Example page since they changed they documentation structure and previous links were outdated already. [a9770b6](https://github.com/scaffold-eth/create-eth-extensions/pull/11/commits/a9770b6ebea101642e16796ffd5f34748d856461)

----
> Notice it has changed it color to violet when you hover over the Switch Icon.

Yes, I mentioned it in the first message. 
> [Tailwind integrations](https://onchainkit.xyz/guides/tailwind) requires to define theme variables, and some of them overwrite ours (works for other projects with tailwind)

I believe it should be fixed on their side, since they [define their color names](https://onchainkit.xyz/guides/tailwind) like `primary`/`secondary`/`success`/`warning`/`error` etc, which will conflict with every project which use the same names.

But for now, I used their colors since it's onchainkit extension 🙂 , and without it their styles will be broken

rin-st commented:
Created an issue https://github.com/coinbase/onchainkit/issues/852

technophile-04 commented on extension/packages/nextjs/layout.tsx.args.mjs:
```diff
@@ -0,0 +1 @@
+export const globalCssImports = 'import "@coinbase/onchainkit/styles.css";';
```
ohh I think this should be `export const imports=` because we renamed the variable in https://github.com/scaffold-eth/create-eth/pull/74/files#diff-d447c6d93270febc35e0430dd2b7dd92fcd104ff952eef1c5c1f753f5a31eb85

rin-st replied:
You're right! Also moved file since it was in wrong folder [6eff85b](https://github.com/scaffold-eth/create-eth-extensions/pull/11/commits/6eff85bf598a11e5356ac0553c5979447132b790)

technophile-04 commented on extension/packages/nextjs/components/ScaffoldEthAppWithProviders.tsx.args.mjs:
```diff
@@ -0,0 +1,4 @@
+export const providerNames = "OnchainKitScaffoldProvider";
+// export const providerSetups = "";
+export const providerImports = `import { OnchainKitScaffoldProvider } from "./OnchainKitScaffoldProvider";`;
+// export const providerProps = "";
```
We could remove the comments?

rin-st commented:
> Also I am not sure whats the best way to handle this but at some places our SE-2 colors are changed for example :

It is fixed in onchainkit `0.26.3`

technophile-04 reviewed (approved):
Tysm @rin-st, this is really nice! and works great!

carletex commented:
Thanks all!! <3

Let's create a PR to main adding this extension + a PR in create-eth to "publish" the curated extension.

rin-st commented:
Oh, forgot about it

> Let's create a PR to main adding this extension 

https://github.com/scaffold-eth/create-eth-extensions/pull/15

> + a PR in create-eth to "publish" the curated extension

It's already added https://github.com/scaffold-eth/create-eth/pull/74/files#diff-7025c91537448e2ff14397d6f1455b06e2e6da125f21beeb860a474a003a65a2R16
